### ISSUE TITLE
EU VAT numbers

### ISSUE THREAD

**Issue by codingjoe** [django/django-localflavor#210]
Hi I'm maintaining `django-vies` a package for european VAT number verification and validation.

I don't use the package myself and like to give it await. Since we are on djangocon it was suggested to merge it into localflavor.

I just did a complete rewrite of the package because I wanted it to be based on a validator rather than on fields only.

It's currently translate in like 3 languages, I'd like to improve that as well. I guess with the help of you guys it should be much easier.

I think there are a couple of different options:
1. You don't want it (which is ok)
2. You only want the validator part, not the model and form fields or widgets
3. You like it all.

I'd like to explain something, that might effect this decision:
The verification is based on regex, which can happen in a validator and works great.
The validation on the other hand is based on the VIES API. Which is great, because it tells you if a number is actually registered but it is not very reliable and should not be executed within a request.

Anyways:
The new code is [HERE](https://github.com/codingjoe/django-vies/tree/version-3)
Feel free to ask me anything, I'm looking forward to here your feedback.

Cheers from DjangoCon EU
Joe

**Comment by benkonrath** [comment]
I just had a chat with @codingjoe at the Django Under the Hood sprints. Here's a summary:

* The EU VAT field is a good fit for localflavor.
* This won't be direct cut and past of the code from django-vies.
* We should use 2 validators: 1 with the syntax validation, 1 with the soap validation. The field will default to only use the syntax validation. Users can opt in to use the soap validation either as part of the normal field validation or manually in a celery task.
* The dependency on suds-jurko will be optional like the PIL/Pillow dep for the ImageField in Django or we will implement the soap validation calls with python directly. codingjoe mentioned that it was a simple call.
* The work will happen on our end, either by a localflavor maintainer (e.g. me) or as a contribution during a sprint. codingjoe can help by providing feedback / reviewing this work.
* It makes sense to use the validation in https://github.com/arthurdejong/python-stdnum instead of having validation rules directly in django-localflavor. Since this is a bit of a change for localflavor, we should post a proposal to the django-dev mailing list before starting to use python-stdnum.

**Comment by claudep** [comment]
FWIW, I'm +1 for depending on python-stdnum instead of rolling our own validation code whenever possible.

**Comment by benkonrath** [comment]
@claudep That's my feeling as well so it's nice to hear you agree.

**Comment by codingjoe** [comment]
I agree that including `python-stdnum` makes sense, this is a separate issue. The packages does provide a lot of functionality that we have already implemented. This introducing it, should also mean removing a lot of functionality.

Furthermore there are already validator functions provided. Maybe it makes sense to introduce a `contrib.django` part into that library adding it here.

Anyhow, but done than perfect ;)
